src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/NormalizedMultiValuesTest.java

```diff
@@ -258,14 +270,44 @@ public void testGetAllNormalizedMultiValuesWfsJson20() throws Exception {
                         + "&outputFormat=application/json";
         JSONObject json = (JSONObject) getAsJSON(request);
         JSONArray features = json.getJSONArray("features");
-        assertEquals(2, features.size());
+        assertEquals(3, features.size());
         // check stations json content
         JSONObject station = getStationById(features, "st.1");
         assertNotNull(station);
         checkStationJson1(station);
         station = getStationById(features, "st.2");
         assertNotNull(station);
         checkStationJson2(station);
+        station = getStationById(features, "st.3");
+        assertNotNull(station);
+        checkStationJson3(station);
+    }
+
+    @Test
+    public void testGetAllNormalizedMultiValuesWfsJsonFormat20() throws Exception {
+        // check if this is an online test with a JDBC based data store
+        if (notJdbcBased()) {
+            // not a JDBC online test
+            return;
+        }
+        // execute the WFS 2.0 request
+        String request =
+                "wfs?request=GetFeature&version=2.0&typename=st_gml32:Station_gml32"
+                        + "&outputFormat=application/json";
+        MockHttpServletResponse response = getAsServletResponse(request);
+
+        String content = response.getContentAsString();
+        validateJsonOutput(content);
+    }
+
+    private void validateJsonOutput(String jsonString) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.enable(JsonParser.Feature.STRICT_DUPLICATE_DETECTION);
+        try {
+            objectMapper.readTree(jsonString);
```

Comment:
I see so you parse the JSON, since no repeated keys are allowed if the parsing passes it means that the repeat keys were encoded as arrays. But still I will add an explicit check validating that we an array with the expected entities inside it.

Reply:
I only checked repeated keys because complete data and structure check is already made on other json tests added on recent PR.